src/app/components/ContactsList.js

```diff
@@ -53,9 +51,16 @@ const ContactsList = ({ contacts, onCheck, history, contactID, location }) => {
         key
     }) => {
         if (canMerge && !index) {
-            return <MergeRow key={key} style={style} />;
+            return (
+                <MergeRow
```

Comment:
I think the `ContactsList` component should take an `onMerge` callback that is defined in the `ContactsContainer`, which would be passed to `MergeRow`, where it'd call with `onMerge(mergeableContacts)`:

1) not have to pass the user keys list
2) be able to use the same function for doing manual marge on checked contacts

Also you should extend the check at https://github.com/ProtonMail/proton-contacts/blob/97436ee76b3b841b5731ce8ba3809f3cb4fb4174/src/app/containers/ContactsContainer.js#L49

to check `!Array.isArray(userKeysList)` so that it doesn't pass here until they are fully loaded

Reply:
The problem that I see with that is that if you want to still display a dynamic progress bar keeping track of the contacts being merged (so that the user sees how much is left in the process), you would need to move the logic for this tracking to `ContactsContainer` too, which IMO would be ugly and not in line with the export/import logic being confined to the export/import modal components.

The other possibility is that there is no dynamic progress bar, and one simply has a loader displayed. Then it would make sense to move the logic for merging outside. But then the user has no way of knowing how advanced the merge process is and IMO it would make for a worse user experience.

Reply:
For the check, I extended it as 
```
if (loadingContactEmails || loadingContacts || loadingUserKeys || loadingContactGroups || !Array.isArray(userKeysList))
```
Although I thought that `loadingUserKeys` would take care of checking that `!Array.isArray(userKeysList) = false` whenever `loadingUserKeys = false`

Reply:
I am just talking about the function that does this here:

```
createModal(<MergeModal contacts={mergeableContacts} userKeysList={userKeysList} />);
```

Reply:
Sorry for my misunderstanding. Yeah, you're totally right!